akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -90,6 +90,10 @@ class Encoder(
               case _: SystemMessageEnvelope ⇒
                 log.error(e, "Failed to serialize system message [{}].", send.message.getClass.getName)
                 throw e
+              case _ if e.isInstanceOf[java.nio.BufferOverflowException] ⇒
+                val reason = new OversizedPayloadException(s"Discarding oversized payload sent to ${send.recipient}: max allowed size ${envelope.byteBuffer.limit()} bytes. Message type [${send.message.getClass.getName}].")
+                log.error(reason, "Transient association error (association remains live)")
```

Comment:
I think it is enough to keep the BufferOverflowException and then and just move the message of "discarding oversized ..." to the log.